lib/standard/text/native.nit

```diff
@@ -46,4 +64,105 @@ extern class NativeString `{ char* `}
 
 	# Parse `self` as a Float.
 	fun atof: Float `{ return atof(self); `}
+
+	# Gets the UTF-8 char at index `pos`
+	#
+	# Index is expressed in Unicode chars
+	#
+	#~~~raw
+	#     assert "かきく".as(FlatString).items.char_at(0) == 'か'
+	#~~~
+	#
+	# If the char at position pos is an invalid Unicode char,
+	# the Unicode replacement character � (0xFFFD) will be used.
+	#
+	#~~~raw
+	#     assert "かきく".as(FlatString).items.char_at(1) == '�'
+	#~~~
+	fun char_at(pos: Int): Char `{
+		char c = self[pos];
+		if((c & 0x80) == 0x00) return (uint32_t)c;
+		if(((c & 0xE0) == 0xC0) && ((self[pos + 1] & 0xC0) == 0x80)) return ((((uint32_t)c) & 0x1F) << 6) + ((((uint32_t)self[pos + 1] & 0x3F)));
+		if(((c & 0xF0) == 0xE0) && ((self[pos + 1] & 0xC0) == 0x80) && ((self[pos + 2] & 0xC0) == 0x80)) return ((((uint32_t)c) & 0xF) << 12) + ((((uint32_t)self[pos + 1]) & 0x3F) << 6) + ((((uint32_t)self[pos + 2] & 0x3F)));
+		if(((c & 0xF7) == 0xF0) && ((self[pos + 1] & 0xC0) == 0x80) && ((self[pos + 2] & 0xC0) == 0x80) && ((self[pos + 3] & 0xC0) == 0x80)) return ((((uint32_t)c) & 0x7) << 18) + ((((uint32_t)self[pos + 1]) & 0x3F) << 12) + ((((uint32_t)self[pos + 2]) & 0x3F) << 6) + ((((uint32_t)self[pos + 3] & 0x3F)));
+		return 0xFFFD;
+	`}
+
+	# Is the byte at position `pos` a valid UTF-8 char start
+	#
+	#~~~raw
+	#       assert "か".items.is_valid_utf8_char(0)
+	#       assert not "か".items.is_valid_utf8_char(1)
```

Comment:
Not sure to understand what you did here.

The "か".items[1] is in fact the second byte of the first char? Since this service is not private, the documentation can be improved.